### PR TITLE
53 duplicate active class on side nav

### LINKED ISSUE #53: Duplicate active class on side nav
screen shot 2018-09-17 at 3 54 51 pm

@joshblack 🙏

### PR DESCRIPTION
**Pull request by dakahn**
Closes #53 

Fixes a bug where there could be duplicate active classes on side nav items if the user selected _Contributing_ and then opened _Getting Started_.

#### Changelog

**New**

- Adds a determineRoutePathIndex function to helpers.
- Adds more verbose comments where necessary.
- Adds a pathPrefix configuration for deployment to gh pages.

**Changed**

- Breaks out existing helper function in SideNavItem component into a new distinct file.
- Minor refactoring to SideNavItem.js.
- Adds `--prefix-paths` flag to `build:external` scripts

**Removed**

- Conditional logic in the SideNavItem component.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (joshblack) [inline .vscode/launch.json]: Could you remove this file and update the `.gitignore` to `.vscode` outside of the files we have checked in there?

File: gatsby-config.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ module.exports = {
   siteMetadata: {
     title: 'Carbon Design System',
   },
+  pathPrefix: '/carbon-website-gatsby',
```
Comment (joshblack) [inline gatsby-config.js]: Should make sure to remove this when merging in 👍

File: package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
   ],
   "scripts": {
     "build": "echo 'ERROR: Please specify environment by running either build:external or build:internal. This will set the correct environment variable for the build.'",
-    "build:external": "rm -rf .cache && cross-env GATSBY_CARBON_ENV=external gatsby build",
+    "build:external": "rm -rf .cache && cross-env GATSBY_CARBON_ENV=external gatsby build --prefix-paths",
```
Comment (joshblack) [inline package.json]: Similar note as above ^

File: src/components/SideNav/SideNavItem.js [hunk 4]
```diff
@@ -15,25 +20,20 @@ export default class SideNavItem extends React.Component {
...
-
-      // Support /components/overview
-      if (pathname.length === 2) {
-        isItemActive = locationContainsPathAtIndex(location, item, -1);
-      }
+  renderSubNavItems = (subItems, location) => {
+    return Object.keys(subItems).map(item => {
+      // Whether or not a side nav "item" is activated. Only navItems that are
+      // routes will ever be active.
+      let isNavItemActive = locationContainsPathAtIndex(
```
Comment (joshblack) [inline src/components/SideNav/SideNavItem.js]: As a style nit, we tend to reserve `let` only for mutable bindings and values. As a result, the tendency is to use `const` by default unless the value changes 👍

Comment (dakahn) [reply src/components/SideNav/SideNavItem.js]: Roger that!

**Comment by alisonjoseph** [comment]
There's ~5px or so whitespace to the side of the nav items that wasn't there before.

screen shot 2018-09-27 at 5 06 47 pm

**Pull request closed without merging**